moler/cmd/at/get_cell_id_nr.py

```diff
@@ -0,0 +1,258 @@
+        Verify with latest 3gpp specification 27.007. Example output:
+
+        +C5GREG: <n>,<stat>[,[<lac>],[<ci>],[<AcT>],[<Allowed_NSSAI_length>], [<Allowed_NSSAI>][,[<cause_type>],
+        [<reject_cause>]][,[<cag_stat>][,<caginfo>]
+
+        OK
+
+        where <n> Possible options for selected_mode, but verify with latest 3gpp specification 27.007.
+
+        :param connection: Moler connection to device, terminal when command is executed.
+        :param prompt: prompt where we start from.
+        :param newline_chars: Characters to split local lines - list.
+        :param runner: Runner to run command.
+        """
+        super(GetCellIdNr, self).__init__(connection, operation='execute', prompt=prompt,
+                                          newline_chars=newline_chars, runner=runner)
+
+    def build_command_string(self):
+        return "AT+C5GREG?"
+
+    def on_new_line(self, line, is_full_line):
+        """
+        Method to parse command output. Will be called after line with command echo.
+
+        +C5GREG: <n>,<stat>[,[<tac>],[<ci>],[<AcT>],[<Allowed_NSSAI_length>],[<Allowed_NSSAI>]...
+                 ...[,<cause_type>,<reject_cause>]][,<cag_stat>][,<caginfo>]
+
+        OK
+
+        Write your own implementation but don't forget to call on_new_line from base class
+
+        :param line: Line to parse, new lines are trimmed
+        :param is_full_line:  False for chunk of line; True on full line (NOTE: new line character removed)
+        :return: None
+        """
+        if is_full_line:
+            try:
+                self._parse_cell_data_nr(line)
+            except ParsingDone:
+                pass
+        return super(GetCellIdNr, self).on_new_line(line, is_full_line)
+
+    # +C5GREG: <raw_output>
+    _re_raw_data = re.compile(r'^\s*\+C5GREG:\s(?P<raw_output>.*).*$')
+
+    def _parse_cell_data_nr(self, line):
+        """
+        Parse network registration status and location information:
+        +C5GREG: 1,11
+        +C5GREG: 1,"54DB","0F6B0578",7,2,"aB"
+        +C5GREG: 2,5,"54DB","0F6B0578",7,2,"aB"
+        +C5GREG: 3,5,,,,,,1,2
+        +C5GREG: 3,5,"54DB","0F6B0578",7,2,"aB",1,2
+        +C5GREG: 4,5,,,,,,1,2
+        +C5GREG: 5,5,"54DB","0F6B0578",7,2,"aB",1,2,123,"info"
+        """
+        results_keys = \
+            ['n', 'stat', 'tac', 'ci', 'AcT', 'Allowed_NSSAI_length', 'Allowed_NSSAI', 'cause_type', 'reject_cause',
+             'cag_stat', 'caginfo']
+        if self._regex_helper.match_compiled(self._re_raw_data, line):
+            string_raw_output = self._regex_helper.groupdict().get('raw_output')
+            self.current_ret['raw_output'] = string_raw_output
+            list_output = string_raw_output.split(',')
+            # Case without <stat> in output pop this value from result keys:
+            if "\"" in list_output[1]:
+                del (results_keys[1])
+            list_output = string_raw_output.split(',')
```

Comment:
duplicated :D - the same in line 87

Reply:
Yes. So far, so good. My bad :(
Corrected. Thanx!